test/unittest/tests/wrappers/config_parser.cpp

```diff
@@ -0,0 +1,53 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *	All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+#include "config_parser.h"
+#include <string>
+
+config_parser::config_parser(const std::map<std::string, std::string> &map)
+    : _config_map(std::move(map)), _cfg{nullptr, nullptr, nullptr}
+{
+}
+
+const std::string &
+config_parser::get_config_value(std::string config) const
+{
+    return _config_map.at(config);
+}
+
+void
+config_parser::insert_config(std::string config, std::string value)
+{
+    _config_map[config] = value;
+}
+
+bool
+config_parser::erase_config(std::string config)
+{
+    auto it = _config_map.find(config);
+    if (it == _config_map.end())
+        return false;
+    _config_map.erase(it);
+    return true;
+}
+
+void
+config_parser::construct_config_string()
```

Comment:
This function probably isn't needed and can be folded into get_config_array. It would make sense to have it if we didn't expect to regenerate the config string every time, and get_config_array only generated it assuming it isn't set, but that isn't the case.